internal/bzlmod/go_deps.bzl

```diff
@@ -57,8 +58,25 @@ def _go_deps_impl(module_ctx):
 
         additional_module_tags = []
         for from_file_tag in module.tags.from_file:
+
+            # We will load all sums from transitively resolved `go.sum` files. We will
+            # only use these after version resolution is completed.
+            for item, new_sum in sums_from_go_mod(module_ctx, from_file_tag.go_mod).items():
+                if item in sums and new_sum != sums[item]:
+                    fail("Multiple mismatching sums for {}@{} found.".format(module_tag.path, sum_version))
+                sums[item] = new_sum
```

Comment:
Could extract these three lines into a function that's reused below to ensure we get the same error message in each case.